apps/costcontrol/js/costcontrol.js

```diff
@@ -617,8 +617,12 @@ var CostControl = (function() {
       result.status = 'success';
       result.data = lastDataUsage;
 
-      costcontrol.lastDataResults = lastDataUsage;
-
+      // Front-end workaround for Bug 1083680: Noticeable difference between by
+      // application breakdown totals and the total displayed in chart and
+      // widget (Exists traffic that cannot be not allocated to an app)
+      if (!perApp) {
```

Comment:
I would preffer to have two caches, one for aggregated query and another one for _by app_ query. What do you think? Although the second cache won't be used, at least we don't miss information.